docs/matchers/regex.md

```diff
@@ -92,11 +92,39 @@ async def remember(opsdroid, config, message):
     await message.respond("OK I'll remember that")
 ```
 
+### Named Groups
+
+Elaborate regular expressions may use many groups, both to capture substrings of interest, as well as to group and structure the regular expression itself. In complex regular expressions, it can become difficult to keep track of the various groups and become unecessarily complex to include multiple groups if you refer to them by their index.  Instead, give the groups names to make it easier to keep track of the different groups and retrieve them later. 
```

Comment:
The command in `... complex regular expressions,` feels weird. I recommend removing it and breaking the sentence into two.